### PR TITLE
drivers: sensor: Add attribute for gain and resolution

### PR DESCRIPTION
**Pull request by Kampi**
**Description**

This PR adds a generic sensor attribute to get / set the gain. See https://github.com/zephyrproject-rtos/zephyr/pull/74547 for the initial discussion

### CODE REVIEW

**Comment by Kampi** [comment]
@MaureenHelm how do we want to continue with this PR?

**Review by MaureenHelm — COMMENTED** [review]
The change looks fine, but it's failing CI

**Comment by Kampi** [comment]
> The change looks fine, but it's failing CI

I´m unable to reproduce the issue. Do you know the reason for the issue or the exact build command? Where can I get this command?

**Comment by Kampi** [comment]
> The change looks fine, but it's failing CI

Hi @MaureenHelm 
I can not reproduce the error. Do you have an idea what´s going wrong here?

File: drivers/sensor/ams/tsl2540/tsl2540.c [hunk 1]
```diff
@@ -17,10 +17,10 @@
 #include <zephyr/sys/byteorder.h>
 #include <zephyr/sys/util.h>
 
-#define TSL2540_INTEGRATION_TIME_MS	(2.81)
-#define TSL2540_DEVICE_FACTOR		(53.0)
+#define TSL2540_INTEGRATION_TIME_MS (2.81)
+#define TSL2540_DEVICE_FACTOR       (53.0)
```
Comment (MaureenHelm) [inline drivers/sensor/ams/tsl2540/tsl2540.c:21]: There are a lot of unrelated whitespace changes like this that add noise to the commit. Please revert them to reduce the diff to only the relevant gain/resolution changes.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by ubieda — APPROVED** [review]

**Review by teburd — APPROVED** [review]
